core/src/main/scala/org/apache/spark/api/python/PythonWorkerFactory.scala

```diff
@@ -110,6 +110,57 @@ private[spark] class PythonWorkerFactory(pythonExec: String, envVars: Map[String
     }
   }
 
+  def createStreamingWorker(): (Socket, Option[Int]) = {
```

Comment:
Nit: Can we add some documentations for this method?

Reply:
Drastically simplified this by reusing existing `createSimpleWorker()`. 
Added brief documentation.